MIP/mip-25/README.md

```diff
@@ -0,0 +1,95 @@
+2. A single or multi signature would be required to acquire temporarily the signing priviledges of aptos framework
+3. Only one authourized key would be able to operate the proxy module and acquire the signing priviledges
+4. The key would be rotated and functionality would be provided in the proxy module to do so
+5. Updates to the framework would use a Move script signed by the approved proxy account to acquire signing priviledges of aptos framework during the scripts execution
+6. Ideally the controlling account would be multisig and this would be a strict requirement for the proxy module.  For test purposes this would be relaxed to a single account
+
+## Reference Implementation
+```rust
+module aptos_framework::proxy {
+    use std::signer;
+    use aptos_framework::account::{create_signer_with_capability, SignerCapability};
+    use aptos_framework::event;
+    use aptos_framework::system_addresses;
+    use aptos_framework::system_addresses::is_aptos_framework_address;
+    friend aptos_framework::genesis;
+
+    struct ProxyController has key {
+        controller: address,
+        aptos_framework_signer_cap: SignerCapability,
+    }
+
+    const EINVALID_CONTROLLER : u64 = 0x1;
+
+    #[event]
+    /// An event triggered when the controller is updated
+    struct ControllerUpdated has store, drop {
+        old_controller: address,
+        new_controller: address,
+    }
+
+    /// Only called during genesis.
+    public(friend) fun initialize(aptos_framework: &signer, aptos_framework_signer_cap: SignerCapability) {
+        system_addresses::assert_aptos_framework(aptos_framework);
+
+        move_to(aptos_framework, ProxyController {
+            controller: @aptos_framework,
+            aptos_framework_signer_cap,
+        })
+    }
+
+    /// Update the controller of the proxy
+    /// Aborts if signed by anyone else except the controller or @aptos_framework
+    /// Emits an event on updating the controller
+    public fun update_controller(signer: &signer, new_controller: address) acquires ProxyController {
+        let proxy_controller = borrow_global_mut<ProxyController>(@aptos_framework);
+
+        assert!(
+            proxy_controller.controller != signer::address_of(signer) ||
+            is_aptos_framework_address(signer::address_of(signer)),
+            EINVALID_CONTROLLER);
+
+        let old_controller = proxy_controller.controller;
+        proxy_controller.controller = new_controller;
+
+        event::emit(
+            ControllerUpdated {
+                old_controller,
+                new_controller,
+            },
+        );
+    }
+
+    #[view]
+    /// Return the controller of the proxy
+    public fun get_controller(): address acquires ProxyController {
+        borrow_global<ProxyController>(@aptos_framework).controller
+    }
+
+    /// Delegate aptos framework's signing cap to proxy
+    public fun delegate_to_proxy(controller: &signer) : signer acquires ProxyController {
```

Comment:
One thing I'm not clear on here, maybe a daft question:

In `initialize` the controller address is set to `@aptos_framework`. So how do we get a non-framework key to call `delegate_to_proxy` in the first place, if we're asserting that the signer of `delegate_to_proxy` is already a controller?

Maybe you could flesh out how to implement "2. A single or multi signature would be required to acquire temporarily the signing priviledges of aptos framework"

Reply:
You would use `update_controller` I believe to set the controller to a new address.

Fleshing out - I think this is `delegate_to_proxy` where a single of multi signature transaction would be used to be delegated proxy to the aptos framework account.

Reply:
So this effectively means you have to do this with the genesis signing key, right? Would we then need to move this up into the genesis transaction?

Reply:
We either initially use aptos governance, which seems overkill, or we would move this into the genesis transaction.  This would only apply for `mainnet` as we control `core_resources` in `devnet` and `testnet` to be able to set the controlller.

Reply:
We would probably want to consider here challenges in removing the proxy later on. That is, if we run Suzuka Genesis with the proxy, that's probably fine. But, we probably don't want to keep it after Suzuka Genesis.

Reply:
I suppose that would be covered in what we migrate to and detailed in a subsequent MIP.  The purpose of this is to proxy Aptos Framework alone, it does not intend to provide a migration plan to how we would govern post Genesis.